### PR TITLE
110 chrome rdp

### LINKED ISSUE #110: Use Chrome's taxonomy of message types
Chrome's [message taxonomy](https://chromedevtools.github.io/debugger-protocol-viewer/1-1/Console/#type-ConsoleMessage) (under ConsoleMessage) is more well structured than ours.

We should be able to convert our packets to a structure more like this in the message utils.

| Theirs |  |
| --- | --- |
| sources | xml, javascript, network, console-api, storage, appcache, rendering, css, security, other, deprecation |
| types | log, dir, dirxml, table, trace, clear, startGroup, startGroupCollapsed, endGroup, assert, timing, profile, profileEnd |
| levels | log, warning, error, debug |

| Ours |  |
| --- | --- |
| type | pageError, consoleAPICall, evaluationResult, reflowActivity, ??? |
| category | CATEGORY_NETWORK, CATEGORY_CSS, CATEGORY_JS, CATEGORY_WEBDEV,CATEGORY_INPUT,CATEGORY_OUTPUT, CATEGORY_SECURITY, CATEGORY_SERVER |
| severity | error, warning, info, log |
| level | error, exception, assert, warn, info, log, clear, trace, table, debug, dir, dirxml, group, groupCollapsed, groupEnd, time, timeEnd, count |

### PR DESCRIPTION
**Pull request by linclark**
Fix #110 

This includes the commits from #115 
## Testing instructions

The messages should all work as before, so I just used the testing page, https://bgrins.github.io/devtools-demos/console/all.html

### CODE REVIEW

**Comment by linclark** [comment]
@bgrins this is ready for review

**Comment by bgrins** [comment]
This is hard to review because it seems to include changes from #115.  Is there a way in the UI to show all commits except for those ones?

**Comment by bgrins** [comment]
@linclark when you have a chance can you rebase this on the latest commits so I can review it?

**Comment by linclark** [comment]
just rebased. I decided not to do the merging of the ConsoleCommand and ConsoleMessage just yet. We may actually want to have those be separate

File: devtools/client/webconsole/new-console-output/components/message-types/page-error.js [hunk 1]
```diff
@@ -23,33 +23,26 @@ PageError.propTypes = {
...
-  const children = [
-    messageBody,
-    repeat
-  ];
 
   // @TODO Use of "is" is a temporary hack to get the category and severity
   // attributes to be applied. There are targeted in webconsole's CSS rules,
   // so if we remove this hack, we have to modify the CSS rules accordingly.
   return dom.div({
-    class: "message cm-s-mozilla",
```
Comment (bgrins) [inline devtools/client/webconsole/new-console-output/components/message-types/page-error.js:40]: Why has the cm-s-mozilla been removed here?  Will this not match the current markup anymore?

File: devtools/client/webconsole/new-console-output/utils/messages.js [hunk 2]
```diff
@@ -23,61 +22,89 @@ const l10n = new WebConsoleUtils.L10n(STRINGS_URI);
...
       }
 
       return new ConsoleMessage({
-        category: CATEGORY_CLASS_FRAGMENTS[CATEGORY_WEBDEV],
-        data,
-        messageType: "ConsoleApiCall",
-        repeatId: getRepeatId(data),
-        severity: SEVERITY_CLASS_FRAGMENTS[LEVELS[data.level]] || "log",
+        source: MESSAGE_SOURCE.CONSOLE_API,
+        type: message.level,
```
Comment (bgrins) [inline devtools/client/webconsole/new-console-output/utils/messages.js]: This should probably be using the local `type` variable which is overridden in certain cases

File: devtools/client/webconsole/new-console-output/utils/messages.js [hunk 3]
```diff
@@ -23,61 +22,89 @@ const l10n = new WebConsoleUtils.L10n(STRINGS_URI);
...
+        level,
+        parameters,
+        messageText,
+        repeatId: getRepeatId(message),
+        category: "console",
```
Comment (bgrins) [inline devtools/client/webconsole/new-console-output/utils/messages.js]: Should this be using a const (i.e. CATEGORY_WEBDEV)

**Comment by bgrins** [comment]
Overall this is looking really good.  Shall we land it ahead of @nchevobbe's work in #101?  Might change some of the property names there but shouldn't hopefully be a big change in relation to that work

File: devtools/client/webconsole/new-console-output/components/message-types/page-error.js (hunk 1 above)
Comment (linclark) [inline devtools/client/webconsole/new-console-output/components/message-types/page-error.js:40]: This is part of the "Match current markup in PageError" commit. It turns out that the markup in PageError did not match the existing markup for page errors.

Comment (bgrins) [inline devtools/client/webconsole/new-console-output/components/message-types/page-error.js:40]: Are you planning to squash when landing?  This seems like it should remain a separate commit / PR in that case

Comment (linclark) [inline devtools/client/webconsole/new-console-output/components/message-types/page-error.js:40]: I can do it as a Part 1/Part 2 deal. I'm not sure how that works with MozReview, but I'll have to figure it out for the console command PR anyway

**Comment by linclark** [comment]
Currently, there shouldn't be any conflict with nchevobbe's work (except for the fact that there's a new debug level that won't get filtered). Since we're leaving the severity in there, he can still rely on that. We can then switch it over to level as part of #121 or its follow up (whichever one removes category/severity).

File: devtools/client/webconsole/new-console-output/components/message-types/page-error.js (hunk 1 above)
Comment (bgrins) [inline devtools/client/webconsole/new-console-output/components/message-types/page-error.js:40]: Hm looking closer I guess this change includes some mixing with the packet changes and markup changes so for bisecting reasons it should either be squashed in, or the markup / class changes should be pulled out into a different PR and landed separately.  I'm OK with squashing if it'll save time

Comment (linclark) [inline devtools/client/webconsole/new-console-output/components/message-types/page-error.js:40]: Is there a reason there shouldn't be a part 1/ part 2? IIRC, this still passes tests when only this commit is applied

Comment (bgrins) [inline devtools/client/webconsole/new-console-output/components/message-types/page-error.js:40]: I was thinking that it wouldn't be displaying errors properly in between (due to packet structure changing and looking for errorMessage instead of messageText)

Comment (linclark) [inline devtools/client/webconsole/new-console-output/components/message-types/page-error.js:40]: Ah, that isn't part of the commit that does the PageError markup changes: https://github.com/devtools-html/gecko-dev/pull/120/commits/8caad3d24c38e47a5d7d7e3d21a6e2734f40684d

**Comment by linclark** [comment]
This landed as eca781da69 and 36998a3b
